### ISSUE TITLE
Unable to round-trip Enums within Map.

### ISSUE THREAD

**Issue by rlubke** [owlike/genson#138]
Consider the following:

```java
@Test
  public void testDeserializationOfEnumInMap() {
    Genson genson = new GensonBuilder().useClassMetadata(true).create();
    Map<String, Object> expected = new LinkedHashMap<>();
    expected.put("minutes", TimeUnit.MINUTES);
    expected.put("seconds", TimeUnit.SECONDS);
    Map result = genson.deserialize(genson.serialize(expected), LinkedHashMap.class);
    assertEquals(expected, result);
  }
```

The equality failure is due to the value within the deserialized result is String vs TimeUnit.

**Comment by EugenCepoi** [comment]
Currently class metadata is being used only for datastructures that are being serialized as a json object. The enum isn't this is why there is no way for Genson to know what the type is when deserializing it back.

I started working some time ago to support class metadata for any type, but this has downsides as the generated json becomes more and more of a ad hoc protocol that wouldn't be friendly to consume by other libs/languages. So I did put it on hold.

**Comment by rlubke** [comment]
Yep, understand the approach.    Perhaps the serialization format could be configurable?
